File: csharp/2015/day02b/Program.cs
using System;
using System.IO;

class Program
{
    static void Usage()
    {
        Console.Error.WriteLine("usage: dotnet run <input file>");
        Environment.Exit(1);
    }

    static int ParseValue(string s)
    {
        if ( !int.TryParse(s, out int value) )
        {
            Console.Error.WriteLine("Error parsing string as number: " + s);
            Environment.Exit(1);
        }
        return value;
    }

    static int Process(string filename)
    {
        int totalLen = 0;

        try
        {
            using ( StreamReader reader = new StreamReader(filename) )
            {
                string? line;
                while ( (line = reader.ReadLine()) != null )
                {
                    string[] parts = line.Split('x');
                    int l = ParseValue(parts[0]);
                    int w = ParseValue(parts[1]);
                    int h = ParseValue(parts[2]);
                    int perim1 = 2 * (l + w);
                    int perim2 = 2 * (l + h);
                    int perim3 = 2 * (w + h);
                    int presentLen = Math.Min(perim1, Math.Min(perim2, perim3));
                    int bowLen = l * w * h;
                    totalLen += presentLen + bowLen;
                }
            }
        }
        catch ( IOException e )
        {
            Console.Error.WriteLine("Error reading file: " + e.Message);
            Environment.Exit(1);
        }

        return totalLen;
    }

    static void Main(string[] args)
    {
        if ( args.Length < 1 )
        {
            Usage();
        }

        string filename = args[0];
        int result = Process(filename);
        Console.WriteLine("result = " + result);
    }
}

File: java/2015/day02b/Program.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Vector;

public class Program {
    public static void usage() {
        System.err.println("usage: java -jar Program.jar <input file>");
        System.exit(1);
    }

    public static int process(String filename) {
        int totalLen = 0;
        try ( BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            while ( (line = reader.readLine()) != null ) {
                String[] numStrs = line.split("x");
                Vector<Integer> dimensions = new Vector<Integer>();
                for ( String numStr : numStrs ) {
                    dimensions.addElement(Integer.parseInt(numStr));
                }
                int perim1 = (dimensions.elementAt(0) + dimensions.elementAt(1)) * 2;
                int perim2 = (dimensions.elementAt(0) + dimensions.elementAt(2)) * 2;
                int perim3 = (dimensions.elementAt(1) + dimensions.elementAt(2)) * 2;
                int presentLen = Integer.min(Integer.min(perim1, perim2), perim3);
                int bowLen = dimensions.elementAt(0) * dimensions.elementAt(1) * dimensions.elementAt(2);
                totalLen += presentLen + bowLen;
            }
        } catch ( IOException e ) {
            System.err.println("Error reading file: " + e.getMessage());
            System.exit(1);
        }
        return totalLen;
    }

    public static final void main(String[] args) {
        if ( args.length < 1 ) {
            usage();
        }
        String filename = args[0];
        int result = process(filename);
        System.out.println("result = " + result);
    }
}
